[Enhancement] Dockerization of anime-sama

Itz-Agasta opened the pull request:
Closes #5 
This PR introduces Docker Compose setup with Separate Services for TUI and Web Modes. 

## Overview
This approach defines separate services in the `docker-compose.yml` file for the two modes (TUI and WEB). Each service uses the same Docker image but overrides the RUN_MODE environment variable to control behavior.

## Benefits
- Two services: web (runs the website) and tui (runs Django shell or other commands).
- Users specify the mode by selecting the desired service in docker-compose-up commands.
- like Build to Run all services:
  ```bash
    docker-compose up --build
  ```
- Run a specific service:
  ```bash
    docker-compose up web
    docker-compose up tui
  ```

### Typo Fix in Scraper.py
A typo in `anime_sama_app/converter/scraper.py` has been corrected. `Original Code (Line 79)` 
In the `else` part of the `f-string`, the string `"... ` was enclosed in double quotes `(")`, causing a syntax error. Strings within f-strings must be enclosed in single quotes `(')` when double quotes are already used for the f-string.

Let me know if you want any changes or improvements. 🥂

Itz-Agasta committed:
Created dockerfile

Itz-Agasta committed:
Updated the scraper.py

There was a typo in the f string.

Itz-Agasta committed:
Created dockerignore

Itz-Agasta committed:
Created docker compose

This is Approach 2: Single Service with Dynamic RUN_MODE
This approach uses a single app service and dynamically sets the RUN_MODE environment variable. By default, it runs in WEB mode, but users can override the mode by setting RUN_MODE before running docker-compose. [explanation](https://github.com/mathisdlg/anime-sama/issues/5#issuecomment-2531658256)

mathisdlg commented:
I think the first approach, which you describe in the issue, is better because in the future I think we will need to add another environment variable to specify the path where to save the downloaded file.
(Sorry I forget to mention the last need in the issue)
PS: Don't forget to request me a review 😉

Itz-Agasta committed:
updated the docker image

Itz-Agasta commented:
> I think the first approach, which you describe in the issue, is better because in the future I think we will need to add another environment variable to specify the path where to save the downloaded file. (Sorry I forget to mention the last need in the issue) PS: Don't forget to request me a review 😉

I have updated the PR to the [1st approach](https://github.com/mathisdlg/anime-sama/issues/5)  & will update the PR description. but I guess I don't have permission to either ask someone for review as well as for adding label in this repo. 
![2024-12-11 (2)](https://github.com/user-attachments/assets/5771dcac-9db5-490d-bf17-721c938c4bf8)

However, feel free to review the pr and let me know if you need any changes or enhancements.

mathisdlg commented:
Why do you use gunicorn and not directly django?
\+ To launch the TUI mode you need to run scrapper.py
(I'm a little bit busy right now but maybe this WE I should find some time to work on this project)

Itz-Agasta committed:
Refactor scraper `main function`

Before at line 132  the main function is defined without parameters, but it is being called with two arguments (index and alias) at line 153.

Itz-Agasta committed:
Updated the dockerfile

1. It now uses django instead of gunicorn for web version.

2. set the TUI mode to scraper.py

Itz-Agasta committed:
Added a Readme.md for docker

Itz-Agasta committed:
Updated the scraper.py

Itz-Agasta commented:
@mathisdlg I just updated the PR as per your feedback. can you review it again when you have time? 
Thanks!

mathisdlg reviewed (approved):
Thanks a lot for the doc

mathisdlg reviewed (approved):
It isn't required but not an issues
